lib/auth/apiserver.go

```diff
@@ -341,6 +349,208 @@ func (s *APIServer) deleteReverseTunnel(auth ClientI, w http.ResponseWriter, r *
 	return message(fmt.Sprintf("reverse tunnel %v deleted", domainName)), nil
 }
 
+type upsertTrustedClusterReq struct {
+	TrustedCluster json.RawMessage `json:"trusted_cluster"`
+}
+
+func (s *APIServer) enableTrustedCluster(auth ClientI, tc services.TrustedCluster, version string) error {
+	vtcr := ValidateTrustedClusterRequest{
+		Token: tc.GetToken(),
+		CAs:   [][]byte{},
```

Comment:
why do you need to marshal here? this request does not seem to go over the wire, so you can pass cert authorities directly to auth server.